client/hotrod-client/src/main/java/org/infinispan/client/hotrod/configuration/NearCacheConfigurationBuilder.java

```diff
@@ -25,6 +28,11 @@ public NearCacheConfigurationBuilder mode(NearCacheMode mode) {
       return this;
    }
 
+   public NearCacheConfigurationBuilder cacheNamePattern(String pattern) {
```

Comment:
I have to admit I am stuck on the fact that the `NearCacheConfiguration` and `NearCacheConfigurationBuilder` classes aren't symmetrical in regards to the type for `cacheNamePattern`. One has a String and the other a Pattern.

I wonder if a user may want to specify a `Pattern` instance with different flags. This might not be that useful though since most of the flags I use for this are related to multiline or dot all, which shouldn't be needed for this case.

We could however have a `cacheNameRegexString` and `cacheNamePattern` method on this class. And the former would just invoke the latter with the compiled `Pattern`. But this is just my thinking, not sure what you think.